From: Werner Koch <wk@gnupg.org>
Date: Mon, 17 Mar 2025 10:06:54 +0100
Subject: [PATCH 02/10] gpgconf: Fix reload and kill of keyboxd.

* tools/gpgconf-comp.c (keyboxd_runtime_change): Fix order of args.
--

Fixes-commit: acaeba2dbdb9bbd68a823c671d5c3577fef5d26d
GnuPG-bug-id: 7569
(cherry picked from commit e212308ede8ecaa3a39ed0366db70da3edfc0ba2)
---
 tools/gpgconf-comp.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tools/gpgconf-comp.c b/tools/gpgconf-comp.c
index fb2a896..f9856ea 100644
--- a/tools/gpgconf-comp.c
+++ b/tools/gpgconf-comp.c
@@ -909,15 +909,15 @@ keyboxd_runtime_change (int killflag)
   int cmdidx;
 
   pgmname = gnupg_module_name (GNUPG_MODULE_NAME_CONNECT_AGENT);
-  argv[i++] = "--no-autostart";
-  argv[i++] = "--keyboxd";
-  cmdidx = i;
-  argv[i++] = killflag? "KILLKEYBOXD" : "RELOADKEYBOXD";
   if (!gnupg_default_homedir_p ())
     {
       argv[i++] = "--homedir";
       argv[i++] = gnupg_homedir ();
     }
+  argv[i++] = "--no-autostart";
+  argv[i++] = "--keyboxd";
+  cmdidx = i;
+  argv[i++] = killflag? "KILLKEYBOXD" : "RELOADKEYBOXD";
   argv[i] = NULL;
   log_assert (i < DIM(argv));
 
